tests/lib/snaps/ovmf/snapcraft.yaml

```diff
@@ -0,0 +1,140 @@
+name: ovmf
+base: core24
+confinement: strict
+version: edk2-stable202411
+summary: EDK2
+description: EDK2
+
+parts:
+  ovmf:
+    build-packages:
+      - uuid-dev
+      - nasm
+      - acpica-tools
+    plugin: nil
+    source: https://github.com/tianocore/edk2.git
+    source-tag: edk2-stable202411
+    override-build: |
+      set +eu
+      . ./edksetup.sh
+      set -eu
+
+      make -C BaseTools/Source/C -j"${CRAFT_PARALLEL_BUILD_COUNT}"
+
+      build_args=(
+        --buildtarget=RELEASE
+        --tagname=GCC5
+        -D TPM1_ENABLE
+        -D TPM2_ENABLE
+        -D TPM2_CONFIG_ENABLE
+        -D SECURE_BOOT_ENABLE
+      )
+      case "${CRAFT_ARCH_BUILD_FOR}" in
+        amd64)
+          build_args+=(
+            --platform='OvmfPkg/OvmfPkgIa32X64.dsc'
+            --arch=IA32
+            --arch=X64
+            -D SMM_REQUIRE
+          )
+          ARCH=X64
+          OVMF=OVMF
+        ;;
+        arm64)
+          build_args+=(
+            --arch=AARCH64
+          )
+          ARCH=AARCH64
+          OVMF=QEMU
+        ;;
+      esac
+      build -n "${CRAFT_PARALLEL_BUILD_COUNT}" \
+        "${build_args[@]}"
+
+      install -Dm644 -t "${CRAFT_PART_INSTALL}/fw" \
+        "Build/Ovmf3264/RELEASE_GCC5/FV/${OVMF}_CODE.fd" \
+        "Build/Ovmf3264/RELEASE_GCC5/FV/${OVMF}_VARS.fd" \
+        "Build/Ovmf3264/RELEASE_GCC5/${ARCH}/Shell.efi"
+
+  import-snakeoil:
+    # We also enroll the snakeoil key for transition. But we should
+    # need to do that.  We can sign our kernels with DB.key instead.
+    plugin: dump
+    source: snakeoil
+    organize:
+      PkKek-1-snakeoil.key: snakeoil/PkKek-1-snakeoil.key
+      PkKek-1-snakeoil.pem: snakeoil/PkKek-1-snakeoil.pem
+    prime:
+      - -*
+
+  efitools:
+    build-packages:
+      - gnu-efi
+      - libssl-dev
+      - sbsigntool
+      - help2man
+      - libfile-slurp-perl
+    after:
+      - import-snakeoil
+    plugin: nil
+    source: https://git.kernel.org/pub/scm/linux/kernel/git/jejb/efitools.git
+    source-tag: v1.9.2
+    override-build: |
+      cp "${CRAFT_STAGE}/snakeoil/PkKek-1-snakeoil.key" snakeoil.key
+      cp "${CRAFT_STAGE}/snakeoil/PkKek-1-snakeoil.pem" snakeoil.crt
+      make -j1 EXTRAKEYS="snakeoil" MYGUID="7bea37ed-f339-470a-aefe-c6b1bef55040"
+      install -Dm644 -t "${CRAFT_PART_INSTALL}/secboot" \
+        LockDown.efi {PK,KEK,DB}.{key,crt}
```

Comment:
this will generate new keys each time during the build which isn't obvious. Probably not a problem but users need to be aware of it. Or we can drop pre-generated keys into the snap.

Reply:
The point here is people not to test explicitly a key. They have to test if a key matches specifically DB or KEK. And not just a constant hash in some test.